### PR TITLE
JFG-993 Jagger2 0 docu

### PR DESCRIPTION
**Pull request by dlatnikov**

### CODE REVIEW

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java [hunk 1]
```diff
@@ -98,9 +117,9 @@ public Builder withValidators(List<Class<? extends ResponseValidator>> validator
         }
 
         /**
-         * As one may expect, creates the object of {@link JLoadTest} type with custom parameters.
+         * Creates the object of JTestDefinition type with custom parameters.
```
Comment (a-badaev) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java]: Why did you remove {@link}?
without it once you rename the referenced class it will not be reflected in a comment

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/aux/MaxLoadThreads.java [hunk 2]
```diff
@@ -2,9 +2,8 @@
 
 /**
  * The maximum number of threads, which Jagger engine can create to provide the requested load.
- * <p>
- * Created by Andrey Badaev
- * Date: 16/11/16
+ * By default it equals 4000. You can change this value in property file
```
Comment (a-badaev) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/aux/MaxLoadThreads.java]: but it is 0 by default in this class

Comment (oskliarov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/aux/MaxLoadThreads.java]: It is not clear: 4000 is maximum allowed value? Or if it is exactly 'Default' value than we need appropriate constructor in JLoadProfileRps

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java [hunk 3]
```diff
@@ -72,24 +89,26 @@ public Builder withQueryProvider(Iterable queryProvider) {
         }
 
         /**
-         * Sets subtypes of {@link com.griddynamics.jagger.invoker.Invoker}.
-         * Instances of this class will be used to during Jagger test execution.
-         * <p/>
+         * Optional: Sets subtypes of {@link com.griddynamics.jagger.invoker.Invoker}
```
Comment (oskliarov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java]: what is default value?

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java [hunk 4]
```diff
@@ -72,24 +89,26 @@ public Builder withQueryProvider(Iterable queryProvider) {
...
+         *
+         * Instances of this class will be used to during Jagger test execution. @n
```
Comment (oskliarov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java]: "... execution" to ... create requests to SUT?

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java [hunk 5]
```diff
@@ -72,24 +89,26 @@ public Builder withQueryProvider(Iterable queryProvider) {
...
             this.invoker = invoker;
             return this;
         }
     
         /**
-         * Sets a list of subtypes of {@link ResponseValidator}
-         * Instances of those subtypes will be used to validate responses during Jagger test execution.
-         * <p/>
+         * Optional: Sets a list of subtypes of {@link com.griddynamics.jagger.engine.e1.collector.ResponseValidator}
+         * Instances of those subtypes will be used to validate responses during Jagger test execution @n
```
Comment (oskliarov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java]: Are there any default validators or default value is empty list?

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java (hunk 1 above)
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java]: doxygen (who is generating our docu) is working better without links. He is generating links to the class names without the link tag

Idea is handling your case perfectly. Refactor->Rename will do the job

Comment (a-badaev) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java]: still think you should have references not just names. there should be cross references in  docs

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java (hunk 5 above)
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java]: empty list at the moment
will specify default values in the different PR
